### PR TITLE
Add support for toplevel JSON API links.

### PR DESCRIPTION
**Pull request by beauby**
As an adapter option:
```ruby
render json: posts, links: { self: { href: "http//api.domain.com/posts" } }
```

### CODE REVIEW

File: lib/active_model/serializable_resource.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'set'
 module ActiveModel
   class SerializableResource
-    ADAPTER_OPTION_KEYS = Set.new([:include, :fields, :adapter, :meta, :meta_key])
+    ADAPTER_OPTION_KEYS = Set.new([:include, :fields, :adapter, :meta, :meta_key, :links])
```
Comment (bf4) [inline lib/active_model/serializable_resource.rb]: nice!  shouldn't this integrate with existing code such as pagination links?

**Comment by NullVoxPopuli** [comment]
:+1:

File: lib/active_model/serializable_resource.rb (hunk 1 above)
Comment (beauby) [inline lib/active_model/serializable_resource.rb]: Not really, pagination is automagically added to resources anyways.

Comment (joaomdmoura) [inline lib/active_model/serializable_resource.rb]: but would both work well together? What do you think about adding a test covering it? :smile:

**Comment by joaomdmoura** [comment]
Just one small comments, would be nice to have this test IMO. Other then this it LGTM

File: lib/active_model/serializable_resource.rb (hunk 1 above)
Comment (beauby) [inline lib/active_model/serializable_resource.rb]: Yes, the pagination stuff just updates the links.

**Comment by joaomdmoura** [comment]
LGTM then, merging it.

**Pull request merged**